capsules/src/console.rs

```diff
@@ -127,27 +141,35 @@ impl<'a, U: UART> Driver for Console<'a, U> {
                                 app.pending_write = true;
                                 app.write_buffer = Some(slice);
                             }
-                            0
+                            ReturnCode::SUCCESS
                         },
-                        None => -1
+                        None => ReturnCode::FAIL
+                        // XXX  ^^^^^^^^^^^^^^^^-- When can we fail to take the write_buffer
+                        //                         which return code is best here?
```

Comment:
I believe this error case is when a user calls subscribe without first calling allow to populate that buffer, right? This is sort of the like `ERESERVE`, but not quite. Thoughts?